File: moderate/pangrams.cs
/* CodeEval - Pangrams
 * https://www.codeeval.com/open_challenges/37/
 */

using System.IO;
using System.Text;
using System.Collections.Generic;

class Program
{
    static string alphabets = "abcdefghijklmnopqrstuvwxyz";

    static string GetMissingAlphas(string string1, string string2)
    {
        string1 = string1.ToLower();        
        string2 = string2.ToLower();
        StringBuilder missingChars = new StringBuilder();
        
        foreach (char letter in string1)
            if (string2.IndexOf(letter) < 0)
                missingChars.Append(letter);

        return missingChars.ToString();
    }


    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            string missingChars = GetMissingAlphas(alphabets, line);
            if (missingChars.Length > 0)
                System.Console.WriteLine(missingChars);
            else 
                System.Console.WriteLine("NULL");
        }
    }
}

File: moderate/pangrams.py
#!/usr/bin/env python2
'''
- CodeEval: Pangrams
- https://www.codeeval.com/open_challenges/37/
'''

from sys import argv

alphabets = "abcdefghijklmnopqrstuvwxyz"

def show_missing_alphas(string1, string2):
    '''return list of chars that are in string1 but not in string2'''
    string2 = string2.lower()
    missing_alphas = [i for i in string1 if i not in string2]
    return missing_alphas


def main():
    with open(argv[1]) as f:
        for line in f:
            missing_alphas = show_missing_alphas(alphabets, line)
            if missing_alphas:
                print( ''.join(missing_alphas) )
            else:
                print("NULL")


if __name__ == "__main__":
    main()
